File: app/src/main/java/com/mss/imagesearcher/view/detail/DetailView.java
package com.mss.imagesearcher.view.detail;

import android.support.annotation.NonNull;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface DetailView extends MvpView {

    @StateStrategyType(value = AddToEndSingleStrategy.class)
    void startLoadImage(@NonNull final String imageURL);

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void showProgress(boolean visible);

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void showSuccessSaveMessage();

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void showFailedSaveMessage();

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void shareImage(@NonNull final String path);

    @StateStrategyType(value = AddToEndSingleStrategy.class)
    void showImage(boolean visible);
}

File: app/src/main/java/com/mss/imagesearcher/view/detail/DetailView.kt
package com.mss.imagesearcher.view.detail

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface DetailView : MvpView {

    @StateStrategyType(value = AddToEndSingleStrategy::class)
    fun startLoadImage(imageURL: String)

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showProgress(visible: Boolean)

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showSuccessSaveMessage()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showFailedSaveMessage()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun shareImage(path: String)

    @StateStrategyType(value = AddToEndSingleStrategy::class)
    fun showImage(visible: Boolean)

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showInfo()
}
